src/conditions/base/evm.ts

```diff
@@ -7,6 +7,39 @@ import { returnValueTestSchema } from './schema';
 
 export const STANDARD_CONTRACT_TYPES = ['ERC20', 'ERC721'];
 
+const functionAbiVariable = Joi.object({
+  internalType: Joi.string().required(),
+  name: Joi.string().required(),
+  type: Joi.string().required(),
+});
+
+const functionAbiSchema = Joi.object({
+  name: Joi.string().required(),
+  type: Joi.string().valid('function').required(),
+  inputs: Joi.array().items(functionAbiVariable),
+  outputs: Joi.array().items(functionAbiVariable),
+  // TODO: Should we restrict this to 'view'?
+  // stateMutability: Joi.string().valid('view').required(),
```

Comment:
That requires Ursula to spend ETH on gas costs, which we almost certainly want to avoid

Reply:
Validating here will only prevent misuse but not bad behavior. If this is a risk, then we should also check the state mutability on Ursula's side.

Reply:
Ursula only performs `eth_call` which is only for `read` contract calls, so presumably the call would fail if an actual transaction was needed. (cc @KPrasch )